lib/ddtrace/opentracer/thread_local_scope.rb

```diff
@@ -0,0 +1,30 @@
+module Datadog
+  module OpenTracer
+    # OpenTracing adapter for thread local scopes
+    class ThreadLocalScope < Scope
+      attr_reader \
+        :finish_on_close
+
+      def initialize(
+        manager:,
+        span:,
+        finish_on_close: true
+      )
+        super(manager: manager, span: span)
+        @finish_on_close = finish_on_close
+        @previous_scope = manager.active
+      end
+
+      # Mark the end of the active period for the current thread and Scope,
+      # updating the ScopeManager#active in the process.
+      #
+      # NOTE: Calling close more than once on a single Scope instance leads to
+      # undefined behavior.
+      def close
+        return unless equal?(manager.active)
+        span.finish if finish_on_close
+        manager.send(:set_scope, @previous_scope)
```

Comment:
There's something pretty icky about this: calling a private method on another class is never desirable. But this more or less how Python was doing it, and we were trying to mirror that implementation.

Ideally, a scope manager would be asked to `close`, then it would call down to this `Scope#close`, not the other way around. But as the API is designed, there's some tight coupling between `Scope` and `ScopeManager` (as evidenced by the `manager` in the constructor`), so I don't know if this is possible.